nginx/config.go

```diff
@@ -186,6 +186,11 @@ http {
 		location / {
 			return 404;
 		}
+		{{ if $routerConfig.statusEndpoint }}
```

Comment:
I would consider moving this section up between the health check and the `location /` block. I know Nginx works by making the best/closest match, but for human readability, it usually makes the most sense to have the more specific patterns listed first so that you can think of unmatched requests as progressively "falling through" the list for a possible match to a more general pattern... if you don't mind.